src/Backend/AMQPBackendDispatcher.php

```diff
@@ -133,7 +133,22 @@ public function getChannel()
             ]);
 
             if ($factory instanceof DelayStrategyAware) {
-                $factory->setDelayStrategy(new RabbitMqDlxDelayStrategy());
+                $delayStrategy = $this->settings['delay_strategy_class'];
+                if (
```

Comment:
Can you please extract this to a method?

Reply:
@core23 You mean the entire: if Aware then do method?

Reply:
Yes, the if statement please.